re-search.c

```diff
@@ -167,10 +167,15 @@ int parse_history() {
 		// sanitize
 		i = CMD_PREFIX_LEN; j = 0;
 		while (i < len) {
-			if (i < (len - 1) && cmdline[i] == '\\' && cmdline[i+1] == '\\') {
+			if (cmdline[i] == '\\' && cmdline[j-1] == '\\') {
```

Comment:
Possible negative index here.

Reply:
You're right. I have added a fixing commit.